src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Index.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Syntax.Nodes;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // Index(source:*, n)
+    internal sealed class IndexFunction : FunctionWithTableInput
+    {
+        public override bool IsSelfContained => true;
+
+        public override bool SupportsParamCoercion => false;
+
```

Comment:
Since we will return specific errors from this function in JS, we must declare it as requiring an error context: `public override bool RequiresErrorContext => true;`.

Reply:
done